scripts/convert_hf_checkpoint.py

```diff
@@ -179,36 +179,33 @@ def copy_weights_phi(
     dtype: Optional[torch.dtype] = None,
 ) -> None:
     weight_map = {
-        "layers.0.wte.weight": "transformer.wte.weight",
-        "layers.{}.ln.bias": "transformer.h.{}.norm_1.bias",
-        "layers.{}.ln.weight": "transformer.h.{}.norm_1.weight",
-        "layers.{}.mixer.Wqkv.bias": "transformer.h.{}.attn.attn.bias",
-        "layers.{}.mixer.Wqkv.weight": "transformer.h.{}.attn.attn.weight",
-        "layers.{}.mixer.out_proj.bias": "transformer.h.{}.attn.proj.bias",
-        "layers.{}.mixer.out_proj.weight": "transformer.h.{}.attn.proj.weight",
-        "layers.{}.mixer.rotary_emb.inv_freq": None,
-        "layers.{}.mlp.fc1.bias": "transformer.h.{}.mlp.fc.bias",
-        "layers.{}.mlp.fc1.weight": "transformer.h.{}.mlp.fc.weight",
-        "layers.{}.mlp.fc2.bias": "transformer.h.{}.mlp.proj.bias",
-        "layers.{}.mlp.fc2.weight": "transformer.h.{}.mlp.proj.weight",
-        f"layers.{config.n_layer + 1}.ln.bias": "transformer.ln_f.bias",
-        f"layers.{config.n_layer + 1}.ln.weight": "transformer.ln_f.weight",
-        f"layers.{config.n_layer + 1}.linear.weight": "lm_head.weight",
-        f"layers.{config.n_layer + 1}.linear.bias": "lm_head.bias",
+        "transformer.embd.wte.weight": "transformer.wte.weight",
+        "transformer.h.{}.ln.bias": "transformer.h.{}.norm_1.bias",
+        "transformer.h.{}.ln.weight": "transformer.h.{}.norm_1.weight",
+        "transformer.h.{}.mixer.Wqkv.bias": "transformer.h.{}.attn.attn.bias",
+        "transformer.h.{}.mixer.Wqkv.weight": "transformer.h.{}.attn.attn.weight",
+        "transformer.h.{}.mixer.out_proj.bias": "transformer.h.{}.attn.proj.bias",
+        "transformer.h.{}.mixer.out_proj.weight": "transformer.h.{}.attn.proj.weight",
+        "transformer.h.{}.mixer.rotary_emb.inv_freq": None,
+        "transformer.h.{}.mlp.fc1.bias": "transformer.h.{}.mlp.fc.bias",
+        "transformer.h.{}.mlp.fc1.weight": "transformer.h.{}.mlp.fc.weight",
+        "transformer.h.{}.mlp.fc2.bias": "transformer.h.{}.mlp.proj.bias",
+        "transformer.h.{}.mlp.fc2.weight": "transformer.h.{}.mlp.proj.weight",
+        "lm_head.ln.weight": "transformer.ln_f.weight",
+        "lm_head.ln.bias": "transformer.ln_f.bias",
+        "lm_head.linear.weight": "lm_head.weight",
+        "lm_head.linear.bias": "lm_head.bias",
     }
 
     for name, param in hf_weights.items():
-        if "layers" in name:
-            from_name, number = layer_template(name, 1)
-            if number in (0, config.n_layer + 1):
-                # these are part of the layers in phi, but not in our implementation
-                to_name = weight_map[name]
-            else:
-                to_name = weight_map[from_name]
-                if to_name is None:
-                    continue
-                # the phi layer numbering is off by 1 compared to ours
-                to_name = to_name.format(number - 1)
+        if name.startswith("layers."):
```

Comment:
Or it's better to have such a check right in the beginning of the function?
Something like:
```python
if any(layer_name.startswith("layers.") for layer_name in hf_weights):
    raise ...
```

Reply:
Yes!